realm/realm-library/src/main/java/io/realm/Realm.java

```diff
@@ -1381,14 +1381,28 @@ public void run() {
     /**
      * Removes all objects of the specified class.
      *
+     * DEPRECATED: Use {@link #delete(Class)} instead.
+     *
      * @param clazz the class which objects should be removed.
      * @throws IllegalStateException if the corresponding Realm is closed or in an incorrect thread.
      */
+    @Deprecated
     public void clear(Class<? extends RealmObject> clazz) {
+        delete(clazz);
+    }
+
+    /**
+     * Deletes all objects of the specified class from the Realm.
+     *
+     * @param clazz the class which objects should be removed.
+     * @throws IllegalStateException if the corresponding Realm is closed or called from an incorrect thread.
+     */
+    public void delete(Class<? extends RealmObject> clazz) {
```

Comment:
I prefer `deleteAll(Class)` instead of `delete(Class)` since this method deletes multiple objects.

What do you think @realm/java?